cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/JpaRestApiClient.java

```diff
@@ -0,0 +1,294 @@
+package uk.gov.homeoffice.digital.sas.cucumberjparest;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Fail.fail;
+import static uk.gov.homeoffice.digital.sas.jparest.controller.enums.RequestParameter.TENANT_ID;
+
+import io.restassured.response.Response;
+import io.restassured.specification.QueryableRequestSpecification;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.SpecificationQuerier;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Objects;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.http.HttpHeaders;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * The purpose of this class is to understand the protocol implemented by the JpaRest library. At
+ * this moment in time it knows how to perform CRUD operations on the endpoints exposed but when
+ * authorisation is implemented in the API this class will be updated to add authentication to the
+ * requests being made.
+ */
+@Component
+public class JpaRestApiClient {
+
+  public static final String API_ROOT_PATH = "/resources/";
+  public static final String TENANT_ID_SYSTEM_PROPERTY_NAME = "cucumber.jparest.tenantId";
```

Comment:
It looks like TENANT_ID_SYSTEM_PROPERTY_NAME is only used to construct  TENANT_ID_SYSTEM_PROPERTY_EL. Should it just be
`private static final String TENANT_ID_SYSTEM_PROPERTY_EL
      = "#{systemProperties['cucumber.jparest.tenantId']}";`

Reply:
This was intentionally written this way to allow referencing the property name as a constant from client code if necessary

Reply:
Not sure if there is a little confusion here, maybe because I used the wrong variable name. I probably should have used TENANT_ID_SYSTEM_PROPERTY. 

I think the only thing that will need to know the expression is the tenantid property that it is used to set.

Reply:
Discussed in huddle and with no strong opinion either way, we opted to not make a change. For discoverability details are already in the README for now.